sway-lsp/src/traverse/typed_tree.rs

```diff
@@ -1231,7 +1231,15 @@ impl<'a> TypedTree<'a> {
         let type_engine = self.ctx.engines.te();
         let decl_engine = self.ctx.engines.de();
         let type_info = type_engine.get(type_id);
-        let symbol_kind = type_info_to_symbol_kind(type_engine, &type_info);
+        // This is necessary because the type engine resolves `Self` & `self` to the type it refers to.
+        // We want to keep the semantics of these keywords.
+        let symbol_kind = if type_span.as_str() == "Self" {
+            SymbolKind::SelfTypeKeyword
+        } else if type_span.as_str() == "self" {
+            SymbolKind::SelfKeyword
```

Comment:
Should this check be moved inside of `type_info_to_symbol_kind`?

Reply:
I did think about doing that but decided to keep the original use case of that function as is. The problem is that TypeInfo during parsing == `TypeInfo::SelfType` but then after type checking it is changed to be the type it refers so, so for example it could now be `TypeInfo::Struct`. This is what was causing the incorrect definitions.

Reply:
On second thought though it can't hurt moving them into this function as it is used elsewhere and guarantees nothing falls through the cracks. I'll update now. thanks :)